tests-trio/test_trio_based_service.py

```diff
@@ -498,3 +498,72 @@ class ServiceTest(Service):
     async with background_trio_service(ServiceTest()) as manager:
         await in_daemon.wait()
         manager.cancel()
+
+
+@pytest.mark.trio
+async def test_trio_service_with_try_finally_cleanup():
+    ready_cancel = trio.Event()
+
+    class TryFinallyService(Service):
+        cleanup_up = False
+
+        async def run(self) -> None:
+            try:
+                ready_cancel.set()
+                await self.manager.wait_finished()
```

Comment:
do we still have the condition that `self.manager.wait_finished` actually never returns inside the `run` of a `Service`? @jannikluhn brought this up elsewhere

Reply:
Yeah, it's the current way to "wait forever" inside of a service.  I believe we do have the proper invariant that this will never return inside of the context of a service as it will always be cancelled before getting to the stage where it can return.